cfme/tests/infrastructure/test_host.py

```diff
@@ -349,13 +351,27 @@ def test_hosts_not_displayed_several_times(appliance, provider, setup_provider):
     assert host_count == navigate_to(appliance.collections.hosts, "All").paginator.items_amount
 
 
-@pytest.fixture
-def setup_provider_min_hosts(request, provider, min_hosts=2):
-    num_hosts = len(provider.data.get('hosts', {}))
-    if num_hosts < min_hosts:
-        pytest.skip(f'Not enough hosts({num_hosts}) to run test. Need at least {min_hosts}')
+@pytest.fixture(params=[1, 2, 'all'])
+def setup_provider_min_hosts(request, provider):
+    if request.param == 1 and "compare" in request._pyfuncitem.name:
+        pytest.skip(f'Compare tests need at least 2 hosts.')
+    elif request.param == "all":
+        min_hosts = len(provider.data.get('hosts', {}))
+        if min_hosts <= 2:
+            pytest.skip(f'"All" hosts case is covered by previous test for {min_hosts} Hosts.')
+        request.param = min_hosts
+    else:
+        min_hosts = request.param
+        num_hosts = len(provider.data.get('hosts', {}))
+        if num_hosts < min_hosts:
+            pytest.skip(f'Not enough hosts({num_hosts}) to run test. Need at least {min_hosts}')
     # Function-scoped fixture to set up a provider
-    return setup_or_skip(request, provider)
+    setup_or_skip(request, provider)
+    hosts_view = navigate_to(provider.collections.hosts, "All")
```

Comment:
Since this is in setup, consider making this check via the REST collection instead of via UI navigation.

```
if len(app.rest_api.collections.hosts.count) < min_hosts:
    pytest.skip()
```

Reply:
I need to figure out how to get the hosts per provider. The api call here returns all hosts on the appliance. I was planning on adding support and parameters to test both appliance and provider collections, so I can use this then.

Reply:
Maybe we should just skip the UI or API check? If we don't have enough hosts in the UI the test will fail. From what I understand the failure would be a good one to catch. Something would be wrong with CF or the framework. So far, I always see the same number of hosts that are in yaml in the UI. We have a bit more overhead because too few hosts via api or UI would be caught after setup, but it sounds like a pretty rare event. Thoughts?